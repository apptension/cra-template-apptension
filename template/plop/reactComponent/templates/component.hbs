import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './{{ camelCase name }}.styles';

export const {{ pascalCase name }} = () => {
  return <Container>{{ pascalCase name }} component</Container>;
};

{{ pascalCase name }}.propTypes = {};
