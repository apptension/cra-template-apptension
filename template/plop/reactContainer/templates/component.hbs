import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './{{ camelCase name }}.styles';

export const {{ pascalCase name }}Component = () => {
  return <Container>{{ pascalCase name }} component</Container>;
};

{{ pascalCase name }}Component.propTypes = {};
