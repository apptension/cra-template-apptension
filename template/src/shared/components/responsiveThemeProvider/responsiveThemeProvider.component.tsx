import React, { useState } from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';
import responsiveTheme from '../../../theme/responsiveTheme';
import { useWindowListener } from '../../hooks/useWindowListener';

const parseTheme = (theme: object): DefaultTheme => responsiveTheme(theme);

export interface ResponsiveThemeProviderProps {
  theme: object;
  children: React.ReactNode;
}

export const ResponsiveThemeProvider = ({ theme: themeDefinition, children }: ResponsiveThemeProviderProps) => {
  const [theme, setTheme] = useState(parseTheme(themeDefinition));
  const handleResize = () => setTheme(parseTheme(themeDefinition));
  useWindowListener('resize', handleResize, { throttle: 200 });

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
