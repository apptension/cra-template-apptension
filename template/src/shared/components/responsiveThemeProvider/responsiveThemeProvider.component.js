import React, { PureComponent, Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import responsiveTheme from '../../../theme/responsiveTheme';
import { WindowListener } from '../windowListener/windowListener.component';

const parseTheme = theme => responsiveTheme(theme);

export class ResponsiveThemeProvider extends PureComponent {
  static propTypes = {
    theme: PropTypes.object,
    children: PropTypes.any,
  };

  state = {
    theme: parseTheme(this.props.theme),
  };

  static getDerivedStateFromProps({ theme }) {
    return { theme: parseTheme(theme) };
  }

  handleResize = () => {
    this.setState({ theme: parseTheme(this.props.theme) });
  };

  render() {
    return (
      <Fragment>
        <WindowListener eventType="resize" throttle={200} onEvent={this.handleResize} />
        <ThemeProvider theme={this.state.theme}>{this.props.children}</ThemeProvider>
      </Fragment>
    );
  }
}
