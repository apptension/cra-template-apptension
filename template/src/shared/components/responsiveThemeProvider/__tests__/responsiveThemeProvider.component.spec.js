import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { ResponsiveThemeProvider } from '../responsiveThemeProvider.component';

describe('ResponsiveThemeProvider: Component', () => {
  const defaultProps = {
    theme: {
      children: 'content',
      theme: {
        foo: 'bar',
      },
    },
  };

  const component = props => <ResponsiveThemeProvider {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });

  describe('theme only contains primitive types', () => {
    it('should pass theme to provider', () => {
      const wrapper = render();
      expect(wrapper.find(ThemeProvider).prop('theme')).toEqual(defaultProps.theme);
    });
  });

  describe('theme contains functions', () => {
    const theme = {
      foo: 'bar',
      fn: () => 100,
    };

    it('should pass function result to theme', () => {
      const wrapper = render({ theme });
      expect(wrapper.find(ThemeProvider).prop('theme')).toEqual({
        foo: 'bar',
        fn: 100,
      });
    });
  });
});
