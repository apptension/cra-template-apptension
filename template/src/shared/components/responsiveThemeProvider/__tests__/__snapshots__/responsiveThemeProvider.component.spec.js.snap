// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveThemeProvider: Component should render correctly 1`] = `
<Fragment>
  <WindowListener
    eventType="resize"
    onEvent={[Function]}
    throttle={200}
  />
  <ThemeProvider
    theme={
      Object {
        "children": "content",
        "theme": Object {
          "foo": "bar",
        },
      }
    }
  />
</Fragment>
`;
